app/src/main/java/io/edanni/movies/ui/activity/MovieListActivity.kt

```diff
@@ -0,0 +1,251 @@
+            this.loadingEmitter = emitter
+        }).debounce(150, TimeUnit.MILLISECONDS).subscribe { (type, swipe, initial) ->
+            Handler(Looper.getMainLooper()).post {
+                if (type == LoadingEventType.START) {
+                    this.noMoviesFoundLayout.visibility = GONE
+                    this.errorLayout.visibility = GONE
+                    if (!swipe && !initial) {
+                        this.bottomProgressBar.visibility = VISIBLE
+                    }
+                    if (initial) {
+                        this.topProgressBar.visibility = VISIBLE
+                    }
+                } else {
+                    this.bottomProgressBar.visibility = INVISIBLE
+                    this.topProgressBar.visibility = INVISIBLE
+                    this.refreshLayout.isRefreshing = false
+                }
+            }
+        }
+
+        if (savedInstanceState == null || !savedInstanceState.containsKey("moviePage")) {
+            loadMovies(initial = true)
+        }
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.search_menu, menu)
+
+        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
+        val searchView = menu!!.findItem(R.id.search).actionView as SearchView
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName))
+        searchView.isSubmitButtonEnabled = false
+        searchView.queryTextChanges()
+                .filter { it.isEmpty() || it.length >= 3 }.debounce(250, TimeUnit.MILLISECONDS)
+                .subscribe { text ->
+                    filter = text.toString()
+                    loadMovies(page = 1, initial = true)
+                }
+        searchView.setOnCloseListener { false }
+        return true
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle?) {
+        super.onRestoreInstanceState(savedInstanceState)
+        moviePage = savedInstanceState?.getSerializable("moviePage") as MoviePage?
+        movieListAdapter.movies = (savedInstanceState?.getSerializable("movies") as MovieList?)?.list!!
+        filter = savedInstanceState?.getString("filter")!!
+        val firstVisibleIndex: Int = savedInstanceState.getInt("firstVisibleIndex")
+        this.gridView.smoothScrollToPosition(firstVisibleIndex)
+    }
+
+    override fun onSaveInstanceState(outState: Bundle?) {
+        outState?.putSerializable("moviePage", moviePage)
+        outState?.putSerializable("movies", MovieList(movieListAdapter.movies))
+        outState?.putString("filter", filter)
+        outState?.putInt("firstVisibleIndex", this.gridView.firstVisiblePosition)
+        super.onSaveInstanceState(outState)
+    }
+
+    private fun showMovieDetail(listMovie: Movie) {
+        if (!loadingMovieDetail) {
+            loadingMovieDetail = true
+            startLoading(initial = true)
+            movieService.getMovieDetails(listMovie.id)
```

Comment:
What happens to the `getMovieDetails` operation if the activity gets destroyed before it finishes?

Reply:
The movie detail activity will open once it finishes. D:

Reading up on the RxJava documentation, the proper way to handle this would be to hold on to the subscription and unsubscribe on the `onDestroy()` method of the movie list activity.